### ISSUE TITLE
Failed to deploy 3.10 when using proxy

### ISSUE THREAD

**Issue by camabeh** [openshift/openshift-ansible#9519]
#### Description

Deploying cluster doesn't work when using proxy (Works on same env without proxy).

##### Version
```
openshift 3.10
ansible 2.6.2
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]
```
```
openshift-ansible-3.10.27-2-53-gdc733c114
```

##### Steps To Reproduce
1. `git clone https://github.com/camabeh/openshift-cluster.git --recursive`
2. `bash ./deploy.local.sh`

##### Expected Results
Successfully deployed minimal openshift cluster.

##### Observed Results
```
TASK [openshift_control_plane : Wait for all control plane pods to become ready] *************************************************************************************************************
ok: [master.192.168.100.10.nip.io] => (item=etcd)
...
failed: [master.192.168.100.10.nip.io] (item=api) => changed=false 
  attempts: 60
  item: api
  msg:
    cmd: /usr/bin/oc get pod master-api-master-local-fix.192.168.100.10.nip.io -o json -n kube-system
    results:
    - {}
    returncode: 1
    stderr: |-
      The connection to the server master.192.168.100.10.nip.io:8443 was refused - did you specify the right host or port?
    stdout: ''
...
failed: [master.192.168.100.10.nip.io] (item=controllers) => changed=false 
  attempts: 60
  item: controllers
  results:
    cmd: /usr/bin/oc get pod master-controllers-master-local-fix.192.168.100.10.nip.io -o json -n kube-system
    results:
    - apiVersion: v1
      kind: Pod
      metadata:
        annotations:
          kubernetes.io/config.hash: a3c3ca56f69ed817bad799176cba5ce8
          kubernetes.io/config.mirror: a3c3ca56f69ed817bad799176cba5ce8
          kubernetes.io/config.seen: '2018-08-10T07:33:57.708356148Z'
          kubernetes.io/config.source: file
          scheduler.alpha.kubernetes.io/critical-pod: ''
        creationTimestamp: '2018-08-10T07:34:32Z'
        labels:
          openshift.io/component: controllers
          openshift.io/control-plane: 'true'
        name: master-controllers-master-local-fix.192.168.100.10.nip.io
        namespace: kube-system
        resourceVersion: '1292'
        selfLink: /api/v1/namespaces/kube-system/pods/master-controllers-master-local-fix.192.168.100.10.nip.io
        uid: d2d2ce79-9c6f-11e8-b4f9-525400c9c704
      spec:
        containers:
        - args:
          - |-
            #!/bin/bash
            set -euo pipefail
            if [[ -f /etc/origin/master/master.env ]]; then
              set -o allexport
              source /etc/origin/master/master.env
            fi
            exec openshift start master controllers --config=/etc/origin/master/master-config.yaml --listen=https://0.0.0.0:8444 --loglevel=${DEBUG_LOGLEVEL:-2}
          command:
          - /bin/bash
          - -c
          image: docker.io/openshift/origin-control-plane:v3.10
          imagePullPolicy: IfNotPresent
          livenessProbe:
            failureThreshold: 3
            httpGet:
              path: healthz
              port: 8444
              scheme: HTTPS
            periodSeconds: 10
            successThreshold: 1
            timeoutSeconds: 1
          name: controllers
          resources: {}
          securityContext:
            privileged: true
          terminationMessagePath: /dev/termination-log
          terminationMessagePolicy: File
          volumeMounts:
          - mountPath: /etc/origin/master/
            name: master-config
          - mountPath: /etc/origin/cloudprovider/
            name: master-cloud-provider
          - mountPath: /etc/containers/registries.d/
            name: signature-import
          - mountPath: /usr/libexec/kubernetes/kubelet-plugins
            mountPropagation: HostToContainer
            name: kubelet-plugins
        dnsPolicy: ClusterFirst
        hostNetwork: true
        nodeName: master-local-fix.192.168.100.10.nip.io
        restartPolicy: Always
        schedulerName: default-scheduler
        securityContext: {}
        terminationGracePeriodSeconds: 30
        tolerations:
        - effect: NoExecute
          operator: Exists
        volumes:
        - hostPath:
            path: /etc/origin/master/
            type: ''
          name: master-config
        - hostPath:
            path: /etc/origin/cloudprovider
            type: ''
          name: master-cloud-provider
        - hostPath:
            path: /etc/containers/registries.d
            type: ''
          name: signature-import
        - hostPath:
            path: /usr/libexec/kubernetes/kubelet-plugins
            type: ''
          name: kubelet-plugins
      status:
        conditions:
        - lastProbeTime: null
          lastTransitionTime: '2018-08-10T07:40:36Z'
          status: 'True'
          type: Initialized
        - lastProbeTime: null
          lastTransitionTime: '2018-08-10T07:47:16Z'
          message: 'containers with unready status: [controllers]'
          reason: ContainersNotReady
          status: 'False'
          type: Ready
        - lastProbeTime: null
          lastTransitionTime: '2018-08-10T07:40:36Z'
          status: 'True'
          type: PodScheduled
        containerStatuses:
        - containerID: docker://a7a92fd4ef6ce30114aea5772b3962123f6ab2081828606f5fa71ce8a3f83043
          image: docker.io/openshift/origin-control-plane:v3.10
          imageID: docker-pullable://docker.io/openshift/origin-control-plane@sha256:8a030a68593d64703c0572454d3fd9475bcfadf5d26d2899f92418516c1c49be
          lastState:
            terminated:
              containerID: docker://a7a92fd4ef6ce30114aea5772b3962123f6ab2081828606f5fa71ce8a3f83043
              exitCode: 2
              finishedAt: '2018-08-10T07:47:15Z'
              reason: Error
              startedAt: '2018-08-10T07:47:00Z'
          name: controllers
          ready: false
          restartCount: 12
          state:
            waiting:
              message: Back-off 5m0s restarting failed container=controllers pod=master-controllers-master-local-fix.192.168.100.10.nip.io_kube-system(a3c3ca56f69ed817bad799176cba5ce8)
              reason: CrashLoopBackOff
        hostIP: 10.0.2.15
        phase: Running
        podIP: 10.0.2.15
        qosClass: BestEffort
        startTime: '2018-08-10T07:40:36Z'
    returncode: 0
  state: list
...
```

If I look at logs from containers:

```
oc get pods --all-namespaces
kube-system   master-api-master-local-fix.192.168.100.10.nip.io           0/1       Running            38         1h
kube-system   master-controllers-master-local-fix.192.168.100.10.nip.io   0/1       CrashLoopBackOff   51         1h
kube-system   master-etcd-master-local-fix.192.168.100.10.nip.io          1/1       Running            6          1h
```

```
oc logs master-api-master-local-fix.192.168.100.10.nip.io
Error from server: Get https://master-local-fix.192.168.100.10.nip.io:10250/containerLogs/kube-system/master-api-master-local-fix.192.168.100.10.nip.io/api: remote error: tls: internal error
oc logs master-etcd-master-local-fix.192.168.100.10.nip.io
Error from server: Get https://master-local-fix.192.168.100.10.nip.io:10250/containerLogs/kube-system/master-etcd-master-local-fix.192.168.100.10.nip.io/etcd: remote error: tls: internal error
oc logs master-controllers-master-local-fix.192.168.100.10.nip.io
Error from server: Get https://master-local-fix.192.168.100.10.nip.io:10250/containerLogs/kube-system/master-controllers-master-local-fix.192.168.100.10.nip.io/controllers: remote error: tls: internal error
```
Journal log:
```
https://gist.github.com/camabeh/7af777dc78e21bea956a366d71499b45
```

##### Additional Information

I use vagrant to spin up virtual machines, it works without a problem, if I don't use proxy.

**Comment by vrutkovs** [comment]
Does setting `openshift_https_proxy` helps in this case?

**Comment by camabeh** [comment]
No, without any change...

**Comment by camabeh** [comment]
If you don't use proxy you can reproduce using squid proxy in docker on the host.

```
docker run --name squid -d -p 3128:3128 datadog/squid
```

setup `http_proxy` in `Vagrantfile` and inventory file in `https://github.com/camabeh/openshift-cluster.git` with `http://192.168.0.106:3128` (hosts IP, example)

**Comment by zoobab** [comment]
I am hitting the same bug with 3.10, any idea how to fix it?

**Comment by flashdumper** [comment]
Same here

**Comment by mshutt** [comment]
We were watching the Kubelet logs (origin-node) and we saw that there were liveness/readiness checks that were failing for some reason.  Then we watched `netstat -anp | egrep 3128`.  3128/tcp is the default Squid cache port of course.  We noted that `hyperkube kubelet` was actually opening connections to the proxy server.  That said, we could not think of a reason that the kubelet would want to or need to go out to the 'net.  We've also been passing the same `NO_PROXY` list since the dawn of time.  It seemed like a bit of a stretch but hey, why not try it...  After all the master-api and master-controllers were basically constantly crash looping and so on...

So, we took `HTTP_PROXY` and `HTTPS_PROXY` out of `/etc/sysconfig/origin-node`.  This env file is read in by systemd prior to starting `hyperkube kubelet`.  We then restarted `origin-node` and voila, everything is awesome!  Except...

...
...
...

Builds (oc start-build bc/${BUILDCONFIGNAME}.

Apparently the build pod gets it's HTTP(S)_PROXY and NO_PROXY env vars from the Kubelet.

I guess more research is required... Is the Kubelet actually ignoring NO_PROXY?  If so, why?
